src/App.tsx

```diff
@@ -1,61 +1,123 @@
 import './App.scss';
+import React, { FormEvent, useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const renderList: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [titleError, setHasTitleError] = useState(false);
```

Comment:
if you store a boolean type - just start name with `is` or `has`
```suggestion
  const [hasTitleError, setHasTitleError] = useState(false);
```